File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven()
    {
        return 40;
    }

    public int RemainingMinutesInOven(int minutesIn)
    {
        return ExpectedMinutesInOven() - minutesIn;
    }

    public int PreparationTimeInMinutes(int nbLayers)
    {
        return nbLayers * 2;
    }

    public int ElapsedTimeInMinutes(int nbLayers, int minutesIn)
    {
        return PreparationTimeInMinutes(nbLayers) + minutesIn;
    }
}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven timeInOven = expectedMinutesInOven - timeInOven

let preparationTimeInMinutes numberOfLayers = 2 * numberOfLayers

let elapsedTimeInMinutes numberOfLayers timeInOven =
    let preparationTime = preparationTimeInMinutes numberOfLayers
    preparationTime + timeInOven
